lib/csv/parser.rb

```diff
@@ -531,6 +532,31 @@ def prepare_separators
       @not_line_end = Regexp.new("[^\r\n]+".encode(@encoding))
     end
 
+    # This method verifies that there are no (obvious) ambiguities with the
+    # provided +col_sep+ and +strip+ parsing options. For example, if +col_sep+
+    # and +strip+ were both equal to +\t+, then there would be no clear way to
+    # parse the input.
+    def validate_strip_and_col_sep_options
+      return unless @strip
+
+      string_condition = @strip.is_a?(String) &&
+        (@column_separator.start_with?(@strip) || @column_separator.end_with?(@strip))
+      if string_condition
+        raise InvalidOptionsError,
+              "The provided strip (#{@escaped_strip}) and " \
+              "col_sep (#{@escaped_column_separator}) options are incompatible."
+      end
+
+      space = [" ", "\t", "\f", "\v"]
+      default_condition = !@strip.is_a?(String) &&
+        (@column_separator.start_with?(*space) || @column_separator.end_with?(*space))
+      if default_condition
+        raise InvalidOptionsError,
+              "The provided strip (true) and " \
+              "col_sep (#{@escaped_column_separator}) options are incompatible."
+      end
```

Comment:
Can we simplify this like the following?

```ruby
if @strip.is_a?(String)
  if @column_separator.start_with?(@strip) || @column_separator.end_with?(@strip)
    raise ...
  end
else
  if Regexp.new("\\A[#{@escaped_strip}]|[#{@escaped_strip}]\\z").match?(@column_separator)
    raise ...
  end
end

Reply:
Yes, I think so. Updated.